File: solutions/python/series/1/series.py
def slices(series, length):
    if length == 0:
        raise ValueError("slice length cannot be zero")
    if length < 0:
        raise ValueError("slice length cannot be negative")
    s_len = len(series)
    if s_len == 0:
        raise ValueError("series cannot be empty")
    if length > s_len:
        raise ValueError("slice length cannot be greater than series length")
    return [series[i:i + length] for i in range(0, s_len - length + 1)]

File: solutions/javascript/series/1/series.js
//
// This is only a SKELETON file for the 'Series' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class Series {
  constructor(series) {
    if(series === "")
      throw new Error('series cannot be empty');
    this.series = series;
  }

  slices(sliceLength) {
    const result = [], l = this.series.length;
    if(sliceLength > l)
      throw new Error('slice length cannot be greater than series length');
    if(sliceLength == 0)
      throw new Error('slice length cannot be zero');
    if(sliceLength < 0)
      throw new Error('slice length cannot be negative');
    for(let i = 0; i + sliceLength <= l; i++) {
      const row = [];
      for(const char of this.series.slice(i, i + sliceLength))
        row.push(Number(char));
      result.push(row);
    }
    return result;
  }
}
